### ISSUE TITLE
(stand alone installation) yum install cadvisor doesn't work in Centos 7

### ISSUE THREAD

**Issue by ameyrk18** [google/cadvisor#1942]
Looks like  yum install cadvisor is failing with the below error. Please see the conflict at the bottom. Looks like something has changed in the centos repo?

Loaded plugins: fastestmirror
base                                                                                                                                                                               | 3.6 kB  00:00:00
docker-ce-stable                                                                                                                                                                   | 2.9 kB  00:00:00
epel/x86_64/metalink                                                                                                                                                               |  13 kB  00:00:00
epel                                                                                                                                                                               | 4.7 kB  00:00:00
extras                                                                                                                                                                             | 3.4 kB  00:00:00
updates                                                                                                                                                                            | 3.4 kB  00:00:00
(1/8): epel/x86_64/group_gz                                                                                                                                                        |  88 kB  00:00:00
(2/8): base/7/x86_64/group_gz                                                                                                                                                      | 166 kB  00:00:00
(3/8): epel/x86_64/updateinfo                                                                                                                                                      | 926 kB  00:00:00
(4/8): docker-ce-stable/x86_64/primary_db                                                                                                                                          |  13 kB  00:00:00
(5/8): extras/7/x86_64/primary_db                                                                                                                                                  | 143 kB  00:00:00
(6/8): updates/7/x86_64/primary_db                                                                                                                                                 | 1.2 MB  00:00:00
(7/8): epel/x86_64/primary_db                                                                                                                                                      | 6.4 MB  00:00:00
(8/8): base/7/x86_64/primary_db                                                                                                                                                    | 5.9 MB  00:00:01
Determining fastest mirrors
 * base: ftp.heanet.ie
 * epel: s3-mirror-eu-west-1.fedoraproject.org
 * extras: mirrors.coreix.net
 * updates: ftp.nluug.nl
Package cadvisor is obsoleted by kubernetes, trying to install kubernetes-1.5.2-0.7.git269f928.el7.x86_64 instead
Resolving Dependencies
--> Running transaction check
---> Package kubernetes.x86_64 0:1.5.2-0.7.git269f928.el7 will be installed
--> Processing Dependency: kubernetes-node = 1.5.2-0.7.git269f928.el7 for package: kubernetes-1.5.2-0.7.git269f928.el7.x86_64
--> Processing Dependency: kubernetes-master = 1.5.2-0.7.git269f928.el7 for package: kubernetes-1.5.2-0.7.git269f928.el7.x86_64
--> Running transaction check
---> Package kubernetes-master.x86_64 0:1.5.2-0.7.git269f928.el7 will be installed
--> Processing Dependency: kubernetes-client = 1.5.2-0.7.git269f928.el7 for package: kubernetes-master-1.5.2-0.7.git269f928.el7.x86_64
---> Package kubernetes-node.x86_64 0:1.5.2-0.7.git269f928.el7 will be installed
--> Processing Dependency: socat for package: kubernetes-node-1.5.2-0.7.git269f928.el7.x86_64
--> Processing Dependency: docker for package: kubernetes-node-1.5.2-0.7.git269f928.el7.x86_64
--> Running transaction check
---> Package docker.x86_64 2:1.13.1-63.git94f4240.el7.centos will be installed
--> Processing Dependency: docker-common = 2:1.13.1-63.git94f4240.el7.centos for package: 2:docker-1.13.1-63.git94f4240.el7.centos.x86_64
--> Processing Dependency: docker-client = 2:1.13.1-63.git94f4240.el7.centos for package: 2:docker-1.13.1-63.git94f4240.el7.centos.x86_64
---> Package kubernetes-client.x86_64 0:1.5.2-0.7.git269f928.el7 will be installed
---> Package socat.x86_64 0:1.7.3.2-2.el7 will be installed
--> Running transaction check
---> Package docker-client.x86_64 2:1.13.1-63.git94f4240.el7.centos will be installed
---> Package docker-common.x86_64 2:1.13.1-63.git94f4240.el7.centos will be installed
--> Processing Dependency: skopeo-containers >= 1:0.1.26-2 for package: 2:docker-common-1.13.1-63.git94f4240.el7.centos.x86_64
--> Processing Dependency: oci-umount >= 2:2.3.3-3 for package: 2:docker-common-1.13.1-63.git94f4240.el7.centos.x86_64
--> Processing Dependency: oci-systemd-hook >= 1:0.1.4-9 for package: 2:docker-common-1.13.1-63.git94f4240.el7.centos.x86_64
--> Processing Dependency: oci-register-machine >= 1:0-5.13 for package: 2:docker-common-1.13.1-63.git94f4240.el7.centos.x86_64
--> Processing Dependency: lvm2 >= 2.02.112 for package: 2:docker-common-1.13.1-63.git94f4240.el7.centos.x86_64
--> Processing Dependency: container-storage-setup >= 0.9.0-1 for package: 2:docker-common-1.13.1-63.git94f4240.el7.centos.x86_64
--> Processing Dependency: container-selinux >= 2:2.51-1 for package: 2:docker-common-1.13.1-63.git94f4240.el7.centos.x86_64
--> Running transaction check
---> Package container-selinux.noarch 2:2.42-1.gitad8f0f7.el7 will be updated
---> Package container-selinux.noarch 2:2.55-1.el7 will be an update
--> Processing Dependency: selinux-policy-targeted >= 3.13.1-183 for package: 2:container-selinux-2.55-1.el7.noarch
--> Processing Dependency: selinux-policy-base >= 3.13.1-183 for package: 2:container-selinux-2.55-1.el7.noarch
--> Processing Dependency: selinux-policy >= 3.13.1-183 for package: 2:container-selinux-2.55-1.el7.noarch
---> Package container-storage-setup.noarch 0:0.9.0-1.rhel75.gite0997c3.el7 will be installed
---> Package lvm2.x86_64 7:2.02.177-4.el7 will be installed
--> Processing Dependency: lvm2-libs = 7:2.02.177-4.el7 for package: 7:lvm2-2.02.177-4.el7.x86_64
--> Processing Dependency: device-mapper-persistent-data >= 0.7.0-0.1.rc6 for package: 7:lvm2-2.02.177-4.el7.x86_64
--> Processing Dependency: liblvm2app.so.2.2(Base)(64bit) for package: 7:lvm2-2.02.177-4.el7.x86_64
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_141)(64bit) for package: 7:lvm2-2.02.177-4.el7.x86_64
--> Processing Dependency: libdevmapper-event.so.1.02(Base)(64bit) for package: 7:lvm2-2.02.177-4.el7.x86_64
--> Processing Dependency: liblvm2app.so.2.2()(64bit) for package: 7:lvm2-2.02.177-4.el7.x86_64
--> Processing Dependency: libdevmapper-event.so.1.02()(64bit) for package: 7:lvm2-2.02.177-4.el7.x86_64
---> Package oci-register-machine.x86_64 1:0-6.git2b44233.el7 will be installed
---> Package oci-systemd-hook.x86_64 1:0.1.15-2.gitc04483d.el7 will be installed
--> Processing Dependency: libyajl.so.2()(64bit) for package: 1:oci-systemd-hook-0.1.15-2.gitc04483d.el7.x86_64
---> Package oci-umount.x86_64 2:2.3.3-3.gite3c9055.el7 will be installed
---> Package skopeo-containers.x86_64 1:0.1.29-3.dev.git7add6fc.el7 will be installed
--> Running transaction check
---> Package device-mapper-event-libs.x86_64 7:1.02.146-4.el7 will be installed
---> Package device-mapper-libs.x86_64 7:1.02.140-8.el7 will be updated
--> Processing Dependency: device-mapper-libs = 7:1.02.140-8.el7 for package: 7:device-mapper-1.02.140-8.el7.x86_64
---> Package device-mapper-libs.x86_64 7:1.02.146-4.el7 will be an update
---> Package device-mapper-persistent-data.x86_64 0:0.7.3-3.el7 will be installed
--> Processing Dependency: libaio.so.1(LIBAIO_0.4)(64bit) for package: device-mapper-persistent-data-0.7.3-3.el7.x86_64
--> Processing Dependency: libaio.so.1(LIBAIO_0.1)(64bit) for package: device-mapper-persistent-data-0.7.3-3.el7.x86_64
--> Processing Dependency: libaio.so.1()(64bit) for package: device-mapper-persistent-data-0.7.3-3.el7.x86_64
---> Package lvm2-libs.x86_64 7:2.02.177-4.el7 will be installed
--> Processing Dependency: device-mapper-event = 7:1.02.146-4.el7 for package: 7:lvm2-libs-2.02.177-4.el7.x86_64
---> Package selinux-policy.noarch 0:3.13.1-166.el7_4.9 will be updated
---> Package selinux-policy.noarch 0:3.13.1-192.el7_5.3 will be an update
--> Processing Dependency: policycoreutils >= 2.5-18 for package: selinux-policy-3.13.1-192.el7_5.3.noarch
---> Package selinux-policy-targeted.noarch 0:3.13.1-166.el7_4.9 will be updated
---> Package selinux-policy-targeted.noarch 0:3.13.1-192.el7_5.3 will be an update
---> Package yajl.x86_64 0:2.0.4-4.el7 will be installed
--> Running transaction check
---> Package device-mapper.x86_64 7:1.02.140-8.el7 will be updated
---> Package device-mapper.x86_64 7:1.02.146-4.el7 will be an update
---> Package device-mapper-event.x86_64 7:1.02.146-4.el7 will be installed
---> Package libaio.x86_64 0:0.3.109-13.el7 will be installed
---> Package policycoreutils.x86_64 0:2.5-17.1.el7 will be updated
--> Processing Dependency: policycoreutils = 2.5-17.1.el7 for package: policycoreutils-python-2.5-17.1.el7.x86_64
---> Package policycoreutils.x86_64 0:2.5-22.el7 will be an update
--> Processing Dependency: libsepol >= 2.5-8 for package: policycoreutils-2.5-22.el7.x86_64
--> Processing Dependency: libselinux-utils >= 2.5-12 for package: policycoreutils-2.5-22.el7.x86_64
--> Running transaction check
---> Package libselinux-utils.x86_64 0:2.5-11.el7 will be updated
---> Package libselinux-utils.x86_64 0:2.5-12.el7 will be an update
--> Processing Dependency: libselinux(x86-64) = 2.5-12.el7 for package: libselinux-utils-2.5-12.el7.x86_64
---> Package libsepol.x86_64 0:2.5-6.el7 will be updated
---> Package libsepol.x86_64 0:2.5-8.1.el7 will be an update
---> Package policycoreutils-python.x86_64 0:2.5-17.1.el7 will be updated
---> Package policycoreutils-python.x86_64 0:2.5-22.el7 will be an update
--> Processing Dependency: setools-libs >= 3.3.8-2 for package: policycoreutils-python-2.5-22.el7.x86_64
--> Processing Dependency: libsemanage-python >= 2.5-9 for package: policycoreutils-python-2.5-22.el7.x86_64
--> Running transaction check
---> Package libselinux.x86_64 0:2.5-11.el7 will be updated
--> Processing Dependency: libselinux(x86-64) = 2.5-11.el7 for package: libselinux-python-2.5-11.el7.x86_64
---> Package libselinux.x86_64 0:2.5-12.el7 will be an update
---> Package libsemanage-python.x86_64 0:2.5-8.el7 will be updated
---> Package libsemanage-python.x86_64 0:2.5-11.el7 will be an update
--> Processing Dependency: libsemanage = 2.5-11.el7 for package: libsemanage-python-2.5-11.el7.x86_64
---> Package setools-libs.x86_64 0:3.3.8-1.1.el7 will be updated
---> Package setools-libs.x86_64 0:3.3.8-2.el7 will be an update
--> Running transaction check
---> Package libselinux-python.x86_64 0:2.5-11.el7 will be updated
---> Package libselinux-python.x86_64 0:2.5-12.el7 will be an update
---> Package libsemanage.x86_64 0:2.5-8.el7 will be updated
---> Package libsemanage.x86_64 0:2.5-11.el7 will be an update
--> Processing Conflict: docker-ce-18.03.0.ce-1.el7.centos.x86_64 conflicts docker
--> Restarting Dependency Resolution with new changes.
--> Running transaction check
---> Package docker-ce.x86_64 0:18.03.0.ce-1.el7.centos will be updated
---> Package docker-ce.x86_64 0:18.03.1.ce-1.el7.centos will be an update
--> Processing Conflict: docker-ce-18.03.1.ce-1.el7.centos.x86_64 conflicts docker-io
--> Processing Conflict: docker-ce-18.03.1.ce-1.el7.centos.x86_64 conflicts docker
--> Finished Dependency Resolution
**Error: docker-ce conflicts with 2:docker-1.13.1-63.git94f4240.el7.centos.x86_64**
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

**Comment by dashpole** [comment]
looks like you already have a different version of docker installed?  I had no idea cAdvisor could be installed with yum...

**Comment by waterplily** [comment]
Hi,

I'm seeing this issue as well.  I'm trying to install onto Centos 7.4 with docker-ce version 18.03.1.  I know Centos 7.5 was released recently, but I'm not ready to upgrade to that.  The cadvisor package version supplied with Centos 7.5 is the same as 7.4 - cadvisor.x86_64 0.4.1-0.3.git6906a8ce.el7 

Trying the --skip-broken workaround still results in nothing getting installed because of dependency issues. 
![image](https://user-images.githubusercontent.com/40176914/41244732-d23d1d1a-6d9d-11e8-9b87-c76c0b023c24.png)

Regards,
Lily

**Comment by dashpole** [comment]
The yum cadvisor RPM is not provided by or managed by this project.  It is possible someone else has added something, but the cAdvisor maintainers do not support it.

**Comment by waterplily** [comment]
@dashpole   
Thank you.  I wasn't aware of that.  I'll try another method of using cadvisor.

**Comment by waterplily** [comment]
I just wanted to provide a further update about the cadvisor RPM, so that the information is visible to anyone that comes across this posted issue.  I found a link that named the authors of the rpm, so I emailed them and asked if the package was going to be maintained further.  I received a response and the answer was that the package has been deprecated and there are no plans to update it further on the RHEL side.

I hope that's useful.
